python/hsfs/core/feature_monitoring_result_engine.py

```diff
@@ -103,3 +164,88 @@ def build_query_params(
+        """Run and upload statistics comparison between detection and reference stats.
+
+        Args:
+            detection_stats_id: int. Id of the detection statistics.
+            detection_stats: Computed statistics for detection data.
+            fm_config: FeatureMonitoringConfig. Feature monitoring configuration.
+            reference_stats_id: int. Id of the reference statistics.
+            reference_stats: Computed statistics for reference data.
+
+        Returns:
+            FeatureMonitoringResult. Feature monitoring result.
+        """
+
+        if reference_stats:
+            difference = self.compute_difference(
+                detection_stats=detection_stats,
+                reference_stats=reference_stats,
+                metric=fm_config.statistics_comparison_config["compare_on"].lower(),
+                relative=fm_config.statistics_comparison_config["relative"],
+            )
+
+            if fm_config.statistics_comparison_config["strict"]:
+                shift_detected = (
+                    True
+                    if difference >= fm_config.statistics_comparison_config["threshold"]
+                    else False
+                )
+            else:
+                shift_detected = (
+                    True
+                    if difference > fm_config.statistics_comparison_config["threshold"]
+                    else False
+                )
+        else:
+            difference = 0
+            shift_detected = False
+
+        return self.save_feature_monitoring_result(
+            config_id=fm_config.id,
+            execution_id=DEFAULT_EXECUTION_ID,
+            detection_stats_id=detection_stats_id,
+            reference_stats_id=reference_stats_id,
+            difference=difference,
+            shift_detected=shift_detected,
+        )
+
+    def compute_difference(
+        self,
+        detection_stats: Dict[Hashable, Any],
+        reference_stats: Dict[Hashable, Any],
+        metric: str,
+        relative: bool = False,
+    ) -> float:
+        """Compute the difference between the reference and detection statistics.
+
+        Args:
+            detection_stats: `Dict[Hashable, Any]`. The statistics computed from detection data.
+            reference_stats: `Dict[Hashable, Any]`. The statistics computed from reference data.
+            metric: `str`. The metric to compute the difference for.
+            relative: `bool`. Whether to compute the relative difference or not.
+
+        Returns:
+            `float`. The difference between the reference and detection statistics.
+        """
+        if reference_stats.get("specific_value", None) is not None:
+            reference_stats[metric] = reference_stats["specific_value"]
+
+        if relative:
+            if (reference_stats[metric] + detection_stats[metric]) == 0:
+                return float("inf")
+            else:
+                return (detection_stats[metric] - reference_stats[metric]) / (
```

Comment:
How to interpret this relative difference? Why not percentage change?

Reply:
Good that you pointed it out, I had forgotten a factor 0.5 in the denominator. It would be the arithmetic mean change, it is quite standard when there is no real reason to use one or the other as a reference for the change. I switch it to the relative difference as you suggest now, I don't think we need to multiply it by a 100 to get a percentage